### PR TITLE
refactor(iot-device): Add log to mqtt transport handler

### PR DESCRIPTION
**Pull request by abhipsaMisra**

### CODE REVIEW

**Review by azabbasi — APPROVED** [review]

File: iothub/device/src/Transport/Mqtt/MqttTransportHandler.cs [hunk 1]
```diff
@@ -634,13 +634,16 @@ public async void OnMessageReceived(Message message)
...
-                            Logging.Error(this, "Received mqtt message on an unrecognized topic, ignoring message. Topic: " + topic);
+                            Logging.Error(this, $"Received mqtt message on an unrecognized topic, ignoring message. Topic: {topic}", nameof(OnMessageReceived));
                         }
                     }
                 }
             }
             catch (Exception ex)
             {
+                if (Logging.IsEnabled)
+                    Logging.Error(this, $"Received an exception while processing an mqtt message: {ex}", nameof(OnMessageReceived));
```
Comment (drwill-ms) [inline iothub/device/src/Transport/Mqtt/MqttTransportHandler.cs:645]: MQTT (all caps)

**Review by drwill-ms — APPROVED** [review]

**Review by jamdavi — APPROVED** [review]

**Pull request merged**